### PR TITLE
actions: Add action to perform checks for pull requests

### PR DESCRIPTION
**Pull request by amshinde**
Use github actions for performing WIP checks on PRs.
The action checks for keywords in subject line
as well labels.

Fixes: #437

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Comment by amshinde** [comment]
@bergwolf @egernst @jodh-intel @sameo PTAL
Once this is merged we can move away from the zuul checks we have in place for this repo and also for the 1.x repos.

cc @ttx

File: .github/workflows/PR-checks.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Pull request checks
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
+      - labeled
+      - unlabeled]
```
Comment (likebreath) [inline .github/workflows/PR-checks.yaml:10]: Looks like a typo, extra dangling `]`?

Comment (amshinde) [reply .github/workflows/PR-checks.yaml]: @likebreath oops, Fixed it!

**Comment by ttx** [comment]
LGTM! When this is ready to merge, we'll need to make the Zuul SoB/WIP checks non-required so that they do not block the merge.

**Comment by ttx** [comment]
I switched the required approvals to the new SoB/WIP checks but this is now blocked by Travis CI failing all tests right now.

**Pull request merged**